packages/salesforcedx-vscode-apex/src/commands/apexTestRunCodeAction.ts

```diff
@@ -153,6 +154,60 @@ export class ApexLibraryTestRunExecutor extends LibraryCommandletExecutor<{}> {
     );
     return new vscode.Range(pos, pos);
   }
+
+  private async mapApexArtifactToFilesystem(
+    tests: ApexTestResultData[],
+    packageDirectories: NamedPackageDir[]
+  ): Promise<Map<string, string>> {
+    // Create a map to store the correlated artifacts
+    const correlatedArtifacts: Map<string, string> = new Map();
+
+    // Iterate over each test in the ApexTestResultData array
+    for (const test of tests) {
+      // Get the name of the test without the extension
+      const testName = test.apexClass.fullName ?? test.apexClass.name;
+      // Add the test name to the set
+      correlatedArtifacts.set(testName, 'unknown');
+    }
+
+    const workspaceFolder = vscode.workspace.workspaceFolders
+      ? vscode.workspace.workspaceFolders[0]
+      : undefined;
+    if (!workspaceFolder) {
+      return correlatedArtifacts;
+    }
+
+    const patterns = packageDirectories.map(pkgDir => {
+      const relativePath = path.relative(
+        workspaceFolder.uri.fsPath,
+        pkgDir.fullPath
+      );
+      return `${relativePath}/**/*.cls`;
+    });
+
+    const findFilesPromises = patterns.map(pattern => {
+      return vscode.workspace.findFiles(pattern, '**/node_modules/**');
+    });
+
+    const filesWithDuplicates = (await Promise.all(findFilesPromises)).flat();
+
+    // Remove duplicates
+    const files = Array.from(
+      new Set(filesWithDuplicates.map(file => file.toString()))
+    ).map(filePath => vscode.Uri.parse(filePath));
+
+    // Iterate over each file found
+    for (const file of files) {
+      // Get the base name of the file without the extension
+      const fileName = path.basename(file.fsPath, '.cls');
+
+      // If the file name is in the testNames set, add it to the correlatedArtifacts map
+      if (correlatedArtifacts.has(fileName)) {
+        correlatedArtifacts.set(fileName, file.fsPath);
+      }
+    }
```

Comment:
fancy stuff.  Just a dumb q. Did you consider just searching for each test class with `vscode.workspace.findFiles`?

Reply:
@gbockus-sf I believe that is what line 188-190 are doing ;)

Reply:
Yeah that does the find for all apex cls files.  We could do a find withe more specific matching for each test file we're looking for.  Then we wouldn't have to parse though all the results.  I'm sort of thinking through this being an issue for larger projects with thousands of cls files.  I'm not sure it's better b/c it if we're doing this for a condition where we have 100s of tests is going to have perf issues the other way (run 100's of finds vs just a single larger find) 

I don't think this is a big concern just figured I'd ask :)

Reply:
I thought about extending the pattern so I would include one pattern for each test. Not sure what the tolerance for findFiles is for pattern length, complexity, etc. I would think that the dir scan would be the slowest part of this whole process and regardless of how fine tuned the patterns are each entry in the directories have to be visited to check for a match.

Reply:
makes sense.